crates/cairo-addons/src/vm/hint_definitions/precompiles.rs

```diff
@@ -0,0 +1,120 @@
+use std::collections::HashMap;
+
+use crate::vm::hint_utils::deserialize_sequence;
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::hint_utils::insert_value_from_var_name,
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::exec_scope::ExecutionScopes,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+use num_bigint::BigUint;
+use num_traits::{ToPrimitive, Zero};
+
+use crate::vm::{
+    hint_utils::{serialize_sequence, Uint256},
+    hints::Hint,
+};
+
+pub const HINTS: &[fn() -> Hint] = &[modexp_gas, modexp_output];
+
+const WORD_SIZE: u32 = 8;
+const MAX_EXP_LEN: u32 = 32;
+const MIN_GAS_COST: u32 = 200;
+const GAS_DIVISOR: u32 = 3;
+
+pub fn modexp_gas() -> Hint {
+    Hint::new(
+        String::from("modexp_gas"),
+        |vm: &mut VirtualMachine,
+         _exec_scopes: &mut ExecutionScopes,
+         ids_data: &HashMap<String, HintReference>,
+         ap_tracking: &ApTracking,
+         _constants: &HashMap<String, Felt252>|
+         -> Result<(), HintError> {
+            let base_length = Uint256::from_var_name("base_length", vm, ids_data, ap_tracking)?;
+            let exp_length = Uint256::from_var_name("exp_length", vm, ids_data, ap_tracking)?;
+            let modulus_length =
+                Uint256::from_var_name("modulus_length", vm, ids_data, ap_tracking)?;
+            let exp_head = Uint256::from_var_name("exp_head", vm, ids_data, ap_tracking)?;
```

Comment:
```suggestion
            let base_length = Uint256::from_var_name("base_length", vm, ids_data, ap_tracking)?.pack();
            let exp_length = Uint256::from_var_name("exp_length", vm, ids_data, ap_tracking)?;
            let modulus_length =
                Uint256::from_var_name("modulus_length", vm, ids_data, ap_tracking)?.pack();
            let exp_head = Uint256::from_var_name("exp_head", vm, ids_data, ap_tracking)?.pack();
```

-> To transform them to BigUints